src/core/src/signature.rs

```diff
@@ -445,14 +445,13 @@ fn default_version() -> f64 {
 }
 
 impl Signature {
-    pub fn name(&self) -> String {
-        if let Some(name) = &self.name {
-            name.clone()
-        } else if let Some(filename) = &self.filename {
-            filename.clone()
-        } else {
-            self.md5sum()
-        }
+    pub fn name(&self) -> Option<String> {
+        self.name.clone()
+    }
+
+    // return name, if not None; or "" if None.
```

Comment:
note: `//` is a comment, `///` is a docstring. This one could also serve as docs, and would show up in https://docs.rs/sourmash/